### ISSUE TITLE
[Conjugation] Possible reference of conjugation/tense labels via a conjugation metadata JSON file

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Android#269]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

A question that's coming to mind right now is how Scribe-Android (and Scribe-iOS) will deal with new conjugations coming in from Scribe-Data via Scribe-Server. Scribe-Data will be getting new data as it's added to Wikidata, and this could include conjugations that have yet to be added into a Scribe keyboard app. This data would then be transferred to a Scribe keyboard application (and Scribe Conjugate applications, for that matter), but then the needed labels for what the conjugation is and which conjugation view should be used would then be unknown by the interface. This would mean that Scribe-Data would be providing new data, but the end applications would need a version update in order to reference it.

One way of handling this would be to include JSON file in the Scribe-Server response that as of now could be hard coded. This metadata file would be loaded into the application and used to define conjugation tenses:
- What's the label of the conjugation tense - i.e. "present simple", "past perfect", etc
- Which fields returned in a verbs response data are included in this tense
- The UI would then be designed in a way that we would not need to hard code the conjugation logic anymore, but rather it would be loaded from the metadata file and the conjugation view would be determined by the number of conjugations in the tense

### Contribution

Happy to discuss this and get feedback on the idea! Once we've decided on the functionality, we can then work to add this file and have the work in #268 be based on it :)

**Comment by andrewtavis** [comment]
@angrezichatterbox: Would be great to get your opinion here :) Let me know if the above isn't clear! Just realized that we're going to have this problem if the data expands and then we don't have a way of remotely updating the fields.

**Comment by andrewtavis** [comment]
CC @wkyoshida, @henrikth93, @mhmohona and @Jag-Marcel as well, as this is a major infrastructure discussion that would affect all Scribe keyboard and conjugate apps 😊

**Comment by angrezichatterbox** [comment]
I am not sure if what I have understood is right or wrong. But we would be constructing the conjugation view based on the JSON response we would be getting from Scribe server. Am I right ?

**Comment by henrikth93** [comment]
I think it would be good to discuss in a meeting maybe? It sounds good, but not sure if I grasp all of it.

**Comment by angrezichatterbox** [comment]
> I think it would be good to discuss in a meeting maybe? It sounds good, but not sure if I grasp all of it.

Yes, that’s true.

**Comment by andrewtavis** [comment]
Yes, let's definitely discuss in the next sync! The big thing is that we might need to be explicit about which fields are in which tense. So say we get a JSON response like:

```json
{
  "wikidata_lexeme_id": {
    "verb_tense_1_conjugation_1": "t_1_conjugation_1",
    "verb_tense_1_conjugation_2": "t_1_conjugation_2",
    "verb_tense_1_conjugation_3": "t_1_conjugation_3",
    "verb_tense_1_conjugation_4": "t_1_conjugation_4",
    "verb_tense_1_conjugation_5": "t_1_conjugation_5",
    "verb_tense_1_conjugation_6": "t_1_conjugation_6",
    "verb_tense_2_conjugation_1": "t_2_conjugation_1",
    "verb_tense_2_conjugation_2": "t_2_conjugation_2",
    "verb_tense_2_conjugation_3": "t_2_conjugation_3",
    "verb_tense_2_conjugation_4": "t_2_conjugation_4",
    "verb_tense_3_conjugation_1": "t_3_conjugation_1",
    ...
  }
}
```

In the above, tense 1 has six conjugations and tense 2 only has four. All of this would be loaded into an SQLite table for verb conjugations, and then till now with Scribe-iOS we hard code which conjugations go in which conjugation view in the app UI. If they change, then the hard coded categorization breaks.

What I'm saying is we provide something like:

```json
{
  "conjugation_view_1": [
    "verb_tense_1_conjugation_1",
    "verb_tense_1_conjugation_2",
    "verb_tense_1_conjugation_3",
    "verb_tense_1_conjugation_4",
    "verb_tense_1_conjugation_5",
    "verb_tense_1_conjugation_6",
  ], 
  "conjugation_view_2": [
    "verb_tense_2_conjugation_1",
    "verb_tense_2_conjugation_2",
    "verb_tense_2_conjugation_3",
    "verb_tense_2_conjugation_4",
  ],
  ...
}
```

In this way we provide the data as well as a contract for how the data should be organized within the UI. In the above the first view is a 3x2 conjugation view, and the second is 2x2. If we update the queries, then we'll update the contract.

There still are some loose ends here, be we do need something like this as if the verb conjugations change and we're hard coding what's in each conjugation view, then we get a bunch of "Not in Wikidata" responses as we're looking for verb conjugation keys that are missing or renamed.

**Comment by andrewtavis** [comment]
Closing this as the decision has been made to use these contracts and the first versions have gone through in #293 :) We'll iterate on these as needed, but the current work will commence in Scribe-Data and then the finalized contracts will be moved over here 😊
